### DISCUSSION TITLE
Built-in rate limiter

### DISCUSSION THREAD

**Discussion by they-cloned-me** [nitrojs/nitro#2165] [Ideas]
I don’t know if this is too far fetched, but it will be great to have a built-in rate limiter, similar to the `express-rate-limit`.
Leveraging the already awesome `routeRules`, we could configure it like:
```ts
routeRules: {
  ‘/blog/**’: {
    rateLimit: {
      max: 60,
      windowMs: 15 * 60 * 1000,
      standardHeaders: true,
      // some other advanced options/configuration
    }  
  },
}
```
I believe doing this will bring users one step closer to the experience they had with frameworks like express.js, while ensuring a wider adoption for nitro/h3.
Personally, this will be super great for the awesome Nuxt framework. 
Right @Atinux, @danielroe? 😉

**Comment by stafyniaksacha** [comment]
Hello

You can use https://nuxt-security.vercel.app/documentation/middleware/rate-limiter or https://github.com/timb-103/nuxt-rate-limit

**Reply by they-cloned-me** [reply]
Hi 👋🏿,
I’ve used those 2 before, plus I had my own custom implementation 😂.
The main reason I opened this issue was so that we could have a standard implementation; not only within Nuxt, but other frameworks like AnalogJS or when Nitro is being used as a standalone api server.
I’ll follow up the escalated issue Pooya forwarded this to.

Thanks.

**Comment by pi0** [comment]
Thanks for suggestion. Escalated to an issue ~> https://github.com/unjs/nitro/issues/2168
